src/tqec/layers/base.py

```diff
@@ -0,0 +1,70 @@
+import numbers
+
+import cirq
+from tqec.detectors.gate import ShiftCoordsGate
+from tqec.generation.circuit import generate_circuit
+from tqec.plaquette.plaquette import Plaquette
+from tqec.position import Position
+from tqec.templates.orchestrator import TemplateOrchestrator
+
+
+def make_repeated_layer(
+    circuit: cirq.Circuit, repetitions: int, add_shift_coord_gate: bool
+) -> cirq.Circuit:
+    circuit_to_repeat = circuit
+    if add_shift_coord_gate:
+        # Note: we do not care on which qubit it is applied, but we want a SHIFT_COORDS instruction
+        #       to be inserted somewhere in the repetition loop. It is inserted at the beginning.
+        any_qubit = next(iter(circuit.all_qubits()), None)
+        assert (
+            any_qubit is not None
+        ), "Could not find any qubit in the given Circuit instance."
```

Comment:
The rationale behind making them `Gate` instances was that a `Detector` does not have to depend on the qubits it is applied on and can be completely independent from them. 
Turns out this was not a good path as:
- the code to make that work is quite messy,
- [`stimcirq`](https://github.com/quantumlib/Stim/blob/main/glue/cirq/stimcirq/_det_annotation.py) is implementing `Detector`s as `Operation` instances.

In the end, I made the wrong decision but succeeded in making the code work quite reliably before realising that my decision was not the best one. Because "working as expected" was sufficient at the time, I did not refactor the code to the best solution.
Now that some parts of the code (this one included) are in consolidation phase, this is definitely something we should try to improve.